lsp/src/text_document.ml

```diff
@@ -159,3 +164,59 @@ let apply_content_changes ?version t changes =
     | Some version -> { document with version }
   in
   { t with document }
+
+type change =
+  { start : int
+  ; stop : int
+  ; replacement : string
+  }
+
+module Edit_map = Map.Make (struct
+  type t = int
+
+  let compare = Int.compare
+end)
+
+let add_edit map change =
+  (* TODO check non overlapping property *)
+  Edit_map.update map ~key:change.start ~f:(function
```

Comment:
how is it that we expect non-overlapping ranges but keep a _multi_-map where the key is the same start offset?

Reply:
We can change the map to use `(start * stop)` as the key.